File: src/utils/logger.js
import { writeTextFile, readTextFile, exists, mkdir } from '@tauri-apps/plugin-fs';

class FrontendLogger {
    constructor() {
        this.logBuffer = [];
        this.logDir = '/Users/ludo/code/ligeia/logs'; // Use absolute path for now
        this.isReady = false;
        this.init();
    }

    async init() {
        return;
        try {
            // Check if logs directory exists, create if not
            const dirExists = await exists(this.logDir);
            if (!dirExists) {
                await mkdir(this.logDir, { recursive: true });
            }
            this.isReady = true;
            
            // Flush any buffered logs
            for (const bufferedLog of this.logBuffer) {
                await this.writeLogToFile(bufferedLog);
            }
            this.logBuffer = [];
        } catch (error) {
            console.error('Failed to initialize frontend logger:', error);
        }
    }

    log(level, component, message, data = null) {
        return;
        const timestamp = new Date().toISOString();
        const logEntry = {
            timestamp,
            level: level.toUpperCase(),
            component,
            message,
            data
        };

        // Always log to console
        const consoleMessage = `[${level.toUpperCase()}] ${component}: ${message}`;
        if (level === 'error') {
            console.error(consoleMessage, data);
        } else if (level === 'warn') {
            console.warn(consoleMessage, data);
        } else if (level === 'debug') {
            console.debug(consoleMessage, data);
        } else {
            console.log(consoleMessage, data);
        }

        // Write to file
        if (this.isReady) {
            this.writeLogToFile(logEntry);
        } else {
            this.logBuffer.push(logEntry);
        }
    }

    async writeLogToFile(logEntry) {
        try {
            const date = new Date().toISOString().split('T')[0];
            const filename = `${this.logDir}/ligeia-frontend-${date}.log`;
            const logLine = JSON.stringify(logEntry) + '\n';
            
            // Append to file
            const fileExists = await exists(filename);
            if (fileExists) {
                // Read existing content and append
                const existingContent = await readTextFile(filename).catch(() => '');
                await writeTextFile(filename, existingContent + logLine);
            } else {
                await writeTextFile(filename, logLine);
            }
        } catch (error) {
            console.error('Failed to write log to file:', error);
        }
    }

    info(component, message, data = null) {
        this.log('info', component, message, data);
    }

    error(component, message, data = null) {
        this.log('error', component, message, data);
    }

    warn(component, message, data = null) {
        this.log('warn', component, message, data);
    }

    debug(component, message, data = null) {
        this.log('debug', component, message, data);
    }
}

// Create global logger instance
const logger = new FrontendLogger();

export default logger;

File: src-tauri/src/logger.rs
use std::fs::{self, OpenOptions};
use std::io::Write;
use std::path::Path;
use chrono::{DateTime, Utc};
use serde_json::json;

pub struct Logger {
    log_dir: String,
}

impl Logger {
    pub fn new() -> Result<Self, Box<dyn std::error::Error>> {
        let log_dir = "../logs".to_string();
        
        // Create logs directory if it doesn't exist
        if !Path::new(&log_dir).exists() {
            fs::create_dir_all(&log_dir)?;
        }
        
        Ok(Logger { log_dir })
    }
    
    pub fn log_info(&self, component: &str, message: &str, data: Option<serde_json::Value>) {
        self.write_log("INFO", component, message, data);
    }
    
    pub fn log_error(&self, component: &str, message: &str, data: Option<serde_json::Value>) {
        self.write_log("ERROR", component, message, data);
        eprintln!("[ERROR] {}: {}", component, message);
    }
    
    pub fn log_debug(&self, component: &str, message: &str, data: Option<serde_json::Value>) {
        self.write_log("DEBUG", component, message, data);
    }
    
    pub fn log_warn(&self, component: &str, message: &str, data: Option<serde_json::Value>) {
        self.write_log("WARN", component, message, data);
    }
    
    fn write_log(&self, level: &str, component: &str, message: &str, data: Option<serde_json::Value>) {
        let timestamp = Utc::now();
        let date_str = timestamp.format("%Y-%m-%d").to_string();
        let log_file = format!("{}/ligeia-backend-{}.log", self.log_dir, date_str);
        
        let log_entry = json!({
            "timestamp": timestamp.to_rfc3339(),
            "level": level,
            "component": component,
            "message": message,
            "data": data
        });
        
        let log_line = format!("{}\n", log_entry.to_string());
        
        // Print to console as well
        println!("[{}] {}: {}", level, component, message);
        
        // Write to file
        if let Ok(mut file) = OpenOptions::new()
            .create(true)
            .append(true)
            .open(&log_file) {
            let _ = file.write_all(log_line.as_bytes());
            let _ = file.flush();
        }
    }
}